test/__init__.py

```diff
@@ -424,6 +427,15 @@ def require_connection(self, func):
             "Cannot connect to MongoDB on %s" % (self.pair,),
             func=func)
 
+    def require_storage_engine(self, engine):
+        """Run a test only if the server is running the specified storage
+        engine (as determined by the `db.serverStatus` command)."""
+        #server_status = self.client.admin.command("serverStatus")
+        #current_engine = server_status["storageEngine"]["name"]
+        return self._require(
+            lambda: engine == self.server_status["storageEngine"]["name"],
```

Comment:
This will fail when `self.server_status` is None. What if we just keep track of the storage engine instead of the entire server status response?